### PR TITLE
talosctl: pin golang

### LINKED ISSUE #264127: talosctl go vendor FOD fails to assemble with go 1.20, worked with 1.19
### Describe the bug
`talosctl` fails to build.

*ERROR*

```console
error: builder for '/nix/store/cf4xm0mgzd6aflkapzapvkkxq47z0za3-talosctl-1.5.4-go-modules.drv' failed with exit code 1;
```
```console
last 50 log lines:
       > go: downloading go.opentelemetry.io/otel/trace v1.10.0
       > go: downloading github.com/google/btree v1.0.1
       > go: downloading github.com/golang-jwt/jwt/v4 v4.5.0
       > go: downloading github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2
       > go: downloading go.etcd.io/etcd/client/v2 v2.305.9
       > go: downloading github.com/munnerz/goautoneg v0.0.0-20191010083416-a7dc8b61c822
       > go: downloading github.com/docker/go-events v0.0.0-20190806004212-e31b211e4f1c
       > go: downloading github.com/go-openapi/swag v0.22.3
       > go: downloading github.com/go-openapi/jsonreference v0.20.2
       > go: downloading go.opencensus.io v0.24.0
       > go: downloading github.com/chai2010/gettext-go v1.0.2
       > go: downloading github.com/MakeNowJust/heredoc v1.0.0
       > go: downloading github.com/exponent-io/jsonpath v0.0.0-20151013193312-d6023ce2651d
       > go: downloading github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7
       > go: downloading github.com/peterbourgon/diskv v2.0.1+incompatible
       > go: downloading github.com/aws/aws-sdk-go-v2/internal/endpoints/v2 v2.4.31
       > go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.4
       > go: downloading github.com/hashicorp/golang-lru v0.5.4
       > go: downloading github.com/blang/semver/v4 v4.0.0
       > go: downloading google.golang.org/appengine v1.6.7
       > go: downloading github.com/mailru/easyjson v0.7.7
       > go: downloading github.com/go-openapi/jsonpointer v0.19.6
       > go: downloading github.com/Azure/go-ansiterm v0.0.0-20210617225240-d185dfc1b5a1
       > go: downloading github.com/moby/spdystream v0.2.0
       > go: downloading golang.org/x/mod v0.12.0
       > go: downloading github.com/go-errors/errors v1.4.2
       > go: downloading github.com/cloudflare/circl v1.3.3
       > go: downloading github.com/go-logr/stdr v1.2.2
       > go: downloading github.com/docker/docker-credential-helpers v0.7.0
       > go: downloading github.com/monochromegane/go-gitignore v0.0.0-20200626010858-205db1a8cc00
       > go: downloading github.com/xlab/treeprint v1.2.0
       > go: downloading github.com/google/shlex v0.0.0-20191202100458-e7afc7fbc510
       > go: downloading github.com/josharian/intern v1.0.0
       > go: downloading go.starlark.net v0.0.0-20230525235612-a134d8f9ddca
       > go: downloading github.com/sethgrid/pester v1.2.0
       > go: downloading github.com/satori/go.uuid v1.2.1-0.20181028125025-b2ce2384e17b
       > go: downloading github.com/spf13/viper v1.14.0
       > go: downloading github.com/rifflock/lfshook v0.0.0-20180920164130-b9218ef580f5
       > go: downloading github.com/gorilla/mux v1.8.0
       > go: downloading github.com/briandowns/spinner v1.19.0
       > go: downloading github.com/mitchellh/mapstructure v1.5.0
       > go: downloading github.com/hashicorp/hcl v1.0.0
       > go: downloading gopkg.in/ini.v1 v1.67.0
       > go: downloading github.com/spf13/cast v1.5.0
       > go: downloading github.com/pelletier/go-toml/v2 v2.0.6
       > go: downloading github.com/magiconair/properties v1.8.7
       > go: downloading github.com/spf13/jwalterweatherman v1.1.0
       > go: downloading github.com/subosito/gotenv v1.4.1
       > go: updates to go.mod needed; to update it:
       >      go mod tidy
```

### Steps To Reproduce
Steps to reproduce the behaviour:
1. `sudo nixos-rebuild switch --impure --upgrade --flake .#MY-HOST
2. 
3. ...

### Expected behavior
I expected the build to be successful.

### Additional context
I had the tool installed on an earlier version. But just decided to update my system.

### Notify maintainers

@flokli 

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
☸ admin@multiverse in nixos on  main [!] took 11s 
λ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.1.58, NixOS, 23.11 (Tapir), 23.11.20231019.7c9cc5a`
 - multi-user?: `yes`
 - sandbox: `relaxed`
 - version: `nix-env (Nix) 2.17.1`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

### PR DESCRIPTION
**Pull request by kirillrdy**
## Description of changes

fixes https://github.com/NixOS/nixpkgs/issues/264127

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by flokli — CHANGES_REQUESTED** [review]
`nix-build -A talosctl` builds for me on nixpkgs master (7484c3dd1ad8c29f4ad3c120c3f356f8e60b45ff), and the commit right before this PR (e00ef8492703d682133f630815ba566e8f72010d) - so this doesn't seem to fix anything.

**Comment by kirillrdy** [comment]
@flokli indeed you are right !

**Comment by kirillrdy** [comment]
looks like what we failed to do last time is check vendored modules

with go 1.21 modules content is not "stable"

nix-build -A talosctl.goModules --check

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -35668,7 +35668,7 @@ with pkgs;
 
   taizen = callPackage ../applications/misc/taizen { };
 
-  talosctl = callPackage ../applications/networking/cluster/talosctl { };
+  talosctl = callPackage ../applications/networking/cluster/talosctl { buildGoModule =  buildGo120Module; };
```
Comment (flokli) [inline pkgs/top-level/all-packages.nix:35671]: Can you add this to a separate line?

Comment (kirillrdy) [reply pkgs/top-level/all-packages.nix:35671]: done

**Review by flokli — APPROVED** [review]

**Pull request merged**